pkg/hive/manager_test.go

```diff
@@ -8,23 +8,38 @@ import (
 	"reflect"
 	"testing"
 
+	"github.com/golang/mock/gomock"
 	hivev1 "github.com/openshift/hive/apis/hive/v1"
-	hivefake "github.com/openshift/hive/pkg/client/clientset/versioned/fake"
 	"github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	kubernetesfake "k8s.io/client-go/kubernetes/fake"
 
 	"github.com/Azure/ARO-RP/pkg/api"
 	"github.com/Azure/ARO-RP/pkg/util/cmp"
+	mock_dynamichelper "github.com/Azure/ARO-RP/pkg/util/mocks/dynamichelper"
+	_ "github.com/Azure/ARO-RP/pkg/util/scheme"
 	"github.com/Azure/ARO-RP/pkg/util/uuid"
 	"github.com/Azure/ARO-RP/pkg/util/uuid/fake"
 	utilerror "github.com/Azure/ARO-RP/test/util/error"
 )
 
+var fakeNamespace = "fake-namespace"
+
+/**
+ * NOTE: I have issue with this...
+ * I really shouldn't be manually stating the type of the group or
+ * specifying its version etc. This is an antipractice imho as I
+ * should be getting this somewhere from the hivev1 library. Not
+ * having it listed there means this is hardcoded and will have to
+ * be manually updated with each version of the library that changes
+ * this thing.
+ */
```

Comment:
Agreed that it's a little frustrating, but I'm not sure of an alternative. We might consider a quick refactor to add a single function that returns the entire `metav1.TypeMeta` object (so everything becomes `TypeMeta: GetClusterDeploymentTypeMeta()`) so we just have 2-3 loc change each time we'd need to update this, but considering it's a `v1` I'm guessing it'll remain stable for a long while.

Reply:
Sounds like a good point, that should make the tests more concise.